packages/components/src/Checkbox.tsx

```diff
@@ -11,41 +11,58 @@ const CheckboxChecked = (props: SVGProps) => (
   </SVG>
 )
 
+const CheckboxIndeterminate = (props: SVGProps) => (
+  <SVG {...props}>
+    <path d="M19,3H5C3.9,3,3,3.9,3,5v14c0,1.1,0.9,2,2,2h14c1.1,0,2-0.9,2-2V5C21,3.9,20.1,3,19,3z M17,13H7v-2h10V13z" />
+  </SVG>
+)
+
 const CheckboxUnchecked = (props: SVGProps) => (
   <SVG {...props}>
     <path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" />
   </SVG>
 )
 
-const CheckboxIcon = (props: SVGProps) => (
-  <React.Fragment>
-    <CheckboxChecked
-      {...props}
-      {...__internalProps({
-        __css: {
-          display: 'none',
-          'input:checked ~ &': {
-            display: 'block',
-          },
-        },
-      })}
-    />
-    <CheckboxUnchecked
-      {...props}
-      {...__internalProps({
-        __css: {
-          display: 'block',
-          'input:checked ~ &': {
+export interface CheckboxIconProps extends SVGProps {
+  isIndeterminate?: boolean
```

Comment:
Thanks for the PR, @dev-cj!

What do you think about renaming this to `indeterminate` to match the attribute on `<input>`?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes

Reply:
Yeah, we should match it. 
Have made the changes.

Reply:
Thanks!